x-pack/plugins/serverless_search/server/lib/indices/fetch_index.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { fetchConnectorByIndexName } from '@kbn/search-connectors';
+
+import { FetchIndexResult } from '../../../common/types';
+
+export async function fetchIndex(
+  client: ElasticsearchClient,
+  indexName: string
+): Promise<FetchIndexResult | undefined> {
+  const [indexData, indexStats, indexCount, connector] = await Promise.all([
```

Comment:
question: What happens if the index doesn't exist or Elasticsearch throws an error for a different reason (temporarily unavailable shards or whatever)?

Reply:
any error in any of these cause with cause this function to throw... which is not exactly ideal.

In practice currently if the index doesn't exist the frontend will never call this endpoint since the index management page will render an error and never render our detail component. BUT that not really good enough for this endpoint to have such poor error handling.

So how should we handle intermittent errors in serverless endpoints? we have utility functions in `enterprise_search` to catch errors from library functions and return errors to the frontend. We haven't gotten that far in serverless search yet.

Reply:
I don't mind just surfacing the underlying Elasticsearch error. The reason we have those obfuscating errors in Search is that Enterprise Search can leak information we don't want to leak, but that's not really an issue for this. 

Could we use `.allSettled()` instead of `.all()` so we return partial information when one call fails?

Reply:
using `.allSettled()` is fine with me, will update.

Reply:
I updated this. My only concern is we're now swallowing errors if any of these calls fail (other than the get index) and we're not logging anything. which is something we should address later.